Gems/Atom/Utils/Code/Include/Atom/Utils/ImGuiMaterialDetails.inl

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <Atom/Utils/ImGuiShaderUtils.h>
+#include <imgui/imgui.h>
+#include <Atom/RPI.Public/Material/Material.h>
+#include <Atom/RPI.Public/MeshDrawPacket.h>
+
+#include <AzCore/Casting/numeric_cast.h>
+
+namespace AZ::Render
+{
+    inline void ImGuiMaterialDetails::SetSelectedDrawPacket(const RPI::MeshDrawPacket* drawPacket)
```

Comment:
There are several large inline functions in this file. Is there any benefit or requirement for them to be inline? If not, it would be preferable to move them to a CPP file.

Reply:
They are in a .inl file, which is #included into the .h file. They must be inline to avoid multiple-definition compiler errors. This is the same pattern used for all ImGui related files in Atom/Utils.